packages/cache/src/TargetHasher.ts

```diff
@@ -18,12 +29,28 @@ export class TargetHasher {
   constructor(private options: TargetHasherOptions) {}
 
   async hash(target: Target): Promise<string> {
+    const { root } = this.options;
+
     const hashKey = await salt(
       target.environmentGlob ?? this.options.environmentGlob ?? ["lage.config.js"],
       `${target.id}|${JSON.stringify(this.options.cliArgs)}`,
       this.options.root,
       this.options.cacheKey || ""
     );
+
+    if (target.cwd === root && target.cache) {
+      if (!target.inputs) {
+        throw new Error("Root-level targets must have `inputs` defined if it has cache enabled.");
+      }
+
+      const hashes = hashGlobGit(target.inputs, { cwd: root, gitignore: false }) ?? {};
```

Comment:
TargetHasher.ts calls hashGlobGit, but so does salt.ts (getEnvHashOneAtTime). I don't totally follow the codepath but it seems like it should be called in one place.

Reply:
So this one is due to the fact that there's a difference between hashing the glob of inputs for a target vs hashing the environment one. This was more important for the per-package tasks. Though the envGlob IS a separate definition than the target input (even if it's a global task)